GREYErrorFormatter.h

```diff
@@ -0,0 +1,66 @@
+//
+// Copyright 2017 Google Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+
+#import <Foundation/Foundation.h>
+
+@class GREYError;
+
+NS_ASSUME_NONNULL_BEGIN
+
+@interface GREYErrorFormatter: NSObject
+
+/**
+ * For a given @c error, create a GREYErrorFormatter to provide a formattedDescription.
+ *
+ * @param error The GREYError used to format the description
+ *
+ * @return An instance of GREYErrorFormatter
+ */
+- (instancetype)initWithError:(GREYError *)error;
+
+/**
+ * Create a human-readable formatted string for the GREYError,
```

Comment:
You can club this with the @return 

/**
 * @return A custom formatting of the GREYError tailored to the type of error.
*/